pkg/reconciler/instances/scmigration/action.go

```diff
@@ -12,11 +14,57 @@ func (a *reconcileAction) Run(ac *service.ActionContext) error {
 	if _, err := ac.KubeClient.Clientset(); err != nil {
 		ac.Logger.Errorf("Failed to retrieve native Kubernetes GO client")
 	}
-
 	ac.Logger.Infof("Action 'reconcileAction' executed (passed version was '%s')", ac.Task.Version)
+	ac.Logger.Infof("Remove owner references from SBUs")
+	cleaner, err := newSCRemovalClient([]byte(ac.KubeClient.Kubeconfig()))
+	if err != nil {
+		return err
+	}
+	if err := cleaner.prepareSBUsForRemoval(ac); err != nil {
+		return err
+	}
+	ac.Logger.Infof("Executing SVCAT => BTP Operator CRD migration")
 	m, err := newMigrator(ac)
 	if err != nil {
 		return err
 	}
-	return m.migrateBTPOperator()
+	if err := m.migrateBTPOperator(); err != nil {
+		return err
+	}
+
+	ac.Logger.Infof("Remove service-catalog")
+	if err := cleaner.removeRelease(ServiceCatalogComponent, ac); err != nil {
+		return err
+	}
+	ac.Logger.Infof("Remove service-catalog-addons")
+	if err := cleaner.removeRelease(ServiceCatalogAddonsComponent, ac); err != nil {
+		return err
+	}
+	ac.Logger.Infof("Remove helm-broker")
+	if err := cleaner.removeRelease(HelmBrokerComponent, ac); err != nil {
+		return err
+	}
+	ac.Logger.Infof("Remove service-manager-proxy")
+	if err := cleaner.removeRelease(ServiceManagerProxyComponent, ac); err != nil {
+		return err
+	}
+
+	time.Sleep(10 * time.Second)
```

Comment:
Any option to verify whether the waiting period was really enough before we continue?

Reply:
I think this was a best-effort guess by the author to reduce flakiness in testing pipelines - https://github.com/kyma-incubator/sc-removal/pull/3. I considered removing the sleeps but then I thought it might be better not to touch the algorithm if it's performing reliably at the moment.

Reply:
after discussing with @piotrmiskiewicz, will remove the sleeps and add more deterministic logic - wait for relevant controllers pods to no longer run. Moving this to draft until I have that implemented.

Reply:
@tobiscr sleeps removed in https://github.com/kyma-incubator/reconciler/pull/722/commits/e58425860a8841e4442cd70b8f4eebd8d1c42039 and replaced by waiting for `helm-broker` and `service-catalog-controller-manager` pods check.